src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbData/breadcrumbDataSummary.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {getMeta} from 'app/components/events/meta/metaProxy';
+import space from 'app/styles/space';
+import {defined} from 'app/utils';
+
+import getBreadcrumbCustomRendererValue from '../../breadcrumbs/getBreadcrumbCustomRendererValue';
+
+type Props = {
+  kvData?: Record<string, any>;
+};
+
+type State = {
+  isExpanded: boolean;
+  hasOverflow: boolean;
+};
+
+class BreadcrumbDataSummary extends React.Component<Props, State> {
+  state = {
+    isExpanded: false,
+    hasOverflow: false,
+  };
+
+  summaryNode = React.createRef<HTMLDivElement>();
+
+  onToggle = () => {
+    this.setState(prevState => ({
+      isExpanded: !prevState.isExpanded,
+    }));
+  };
+
+  renderData = () => {
+    const {kvData} = this.props;
+
+    if (!kvData) {
+      return null;
+    }
+
+    return Object.keys(kvData)
+      .filter(key => defined(kvData[key]) && !!kvData[key])
+      .map(key => (
+        <BreadcrumbDataSummaryData key={key}>
+          <BreadcrumbDataSummaryDataLabel>{`${key}: `}</BreadcrumbDataSummaryDataLabel>
+          <StyledPre>
+            {getBreadcrumbCustomRendererValue({
+              value:
+                typeof kvData[key] === 'object'
+                  ? JSON.stringify(kvData[key])
+                  : kvData[key],
+              meta: getMeta(kvData, key),
+            })}
+          </StyledPre>
+        </BreadcrumbDataSummaryData>
+      ));
+  };
+
+  // TODO(Priscila): implement Summary lifecycles
+  render() {
+    const {children} = this.props;
+    return (
+      <div>
+        <div onClick={this.onToggle} ref={this.summaryNode}>
+          <StyledPre>
+            <code>{children}</code>
+          </StyledPre>
+        </div>
+        {this.renderData()}
+      </div>
+    );
+  }
+}
+
+export default BreadcrumbDataSummary;
+
+const StyledPre = styled('pre')`
```

Comment:
I feel like I've seen this component name/CSS before in this pull request should there be a reusable component for this?

Reply:
I've changed in the current breadcrumb and in the new version..that is the reason you have seen this twice... in the future it's gonna be only one ..so no need to make it reusable 😉